src/portable/portable.c

```diff
@@ -588,9 +617,36 @@ static int extract_image_and_extensions(
                 }
 
                 STRV_FOREACH(p, extension_image_paths) {
+                        _cleanup_(pick_result_done) PickResult ext_result = PICK_RESULT_NULL;
                         _cleanup_(image_unrefp) Image *new = NULL;
+                        const char *path = *p;
+
+                        if (path_is_absolute(*p)) {
+                                r = path_pick(/* toplevel_path= */ NULL,
+                                              /* toplevel_fd= */ AT_FDCWD,
+                                              *p,
+                                              &pick_filter_image_raw,
+                                              PICK_ARCHITECTURE|PICK_TRIES|PICK_RESOLVE,
+                                              &ext_result);
+                                if (r == -EISDIR)
+                                        r = path_pick(/* toplevel_path= */ NULL,
+                                                      /* toplevel_fd= */ AT_FDCWD,
+                                                      *p,
+                                                      &pick_filter_image_dir,
+                                                      PICK_ARCHITECTURE|PICK_TRIES|PICK_RESOLVE,
+                                                      &ext_result);
+                                if (r < 0)
+                                        return r;
+                                if (!ext_result.path)
+                                        return log_debug_errno(
+                                                        SYNTHETIC_ERRNO(ENOENT),
+                                                        "No matching entry in .v/ directory %s found.",
+                                                        *p);
+
+                                path = ext_result.path;
+                        }
 
-                        r = image_find_harder(IMAGE_PORTABLE, *p, NULL, &new);
+                        r = image_find_harder(IMAGE_PORTABLE, path, NULL, &new);
```

Comment:
vpick pins the fd of the selected image, so maybe use `image_from_path(FORMAT_PROC_FD_PATH(result.fd, ...)` if vpick is used?

Reply:
Doesn't work, the actual image name is needed to be parsed, it stops working with that change